File: old_backend/important-core-code/go/vulnerable_path.go
// INTENTIONALLY VULNERABLE CODE - FOR SECURITY SCANNER TESTING ONLY
// This file contains critical path traversal vulnerabilities for testing Aikido scanner

package main

import (
    "io/ioutil"
    "net/http"
)

func downloadHandler(w http.ResponseWriter, r *http.Request) {
    filename := r.URL.Query().Get("file")
    
    // VULNERABLE: No path validation
    content, _ := ioutil.ReadFile("/var/data/" + filename)
    w.Write(content)
}

func serveHandler(w http.ResponseWriter, r *http.Request) {
    path := r.URL.Query().Get("path")
    
    // VULNERABLE: Direct file serving
    http.ServeFile(w, r, path)
}

func main() {
    http.HandleFunc("/download", downloadHandler)
    http.HandleFunc("/serve", serveHandler)
    http.ListenAndServe(":8080", nil)
}

File: old_backend/important-core-code/python/vulnerable_path.py
# INTENTIONALLY VULNERABLE CODE - FOR SECURITY SCANNER TESTING ONLY
# This file contains critical path traversal vulnerabilities for testing Aikido scanner

from flask import Flask, request, send_file
import os

app = Flask(__name__)

@app.route('/download')
def download():
    filename = request.args.get('file')
    # VULNERABLE: No path validation
    return send_file(f'/var/data/{filename}')

@app.route('/read')
def read_file():
    path = request.args.get('path')
    # VULNERABLE: Direct file access
    with open(path, 'r') as f:
        return f.read()

@app.route('/static')
def serve_static():
    resource = request.args.get('resource')
    # VULNERABLE: os.path.join doesn't prevent traversal
    filepath = os.path.join('/app/static', resource)
    return send_file(filepath)

if __name__ == '__main__':
    app.run()
